src/lang/modifyAst/addFillet.test.ts

```diff
@@ -439,6 +441,115 @@ const extrude001 = extrude(50, sketch001)
+  }
+  const ast = astOrError as Program
+
+  // selection
+  const segmentRanges: Array<[number, number]> = selectionSnippets.map(
+    (selectionSnippet) => [
+      code.indexOf(selectionSnippet),
+      code.indexOf(selectionSnippet) + selectionSnippet.length,
+    ]
+  )
+  const selection: Selections = {
+    codeBasedSelections: segmentRanges.map((segmentRange) => ({
+      range: segmentRange,
+      type: 'default',
+    })),
+    otherSelections: [],
+  }
+
+  // radius
+  const radius: KclCommandValue = {
+    valueAst: createLiteral(radiusValue) as Expr,
+    valueText: radiusValue.toString(),
+    valueCalculated: radiusValue.toString(),
+  }
+
+  // programMemory and artifactGraph
+  await kclManager.executeAst({ ast })
+
+  // apply fillet to selection
+  const result = modifyAstWithFilletAndTag(ast, selection, radius)
+  if (err(result)) {
+    return result
+  }
+  const { modifiedAst } = result
+
+  const newCode = recast(modifiedAst)
+
+  expect(newCode).toContain(expectedCode)
+}
+describe('Testing applyFilletToSelection', () => {
+  it('should add a fillet to a specific segment after extrusion', async () => {
+    const code = `const sketch001 = startSketchOn('XY')
+  |> startProfileAt([-10, 10], %)
+  |> line([20, 0], %)
+  |> line([0, -20], %)
+  |> line([-20, 0], %)
+  |> lineTo([profileStartX(%), profileStartY(%)], %)
+  |> close(%)
+const extrude001 = extrude(-15, sketch001)`
+    const segmentSnippets = ['line([0, -20], %)']
+    const radiusValue = 3
+    const expectedCode = `const sketch001 = startSketchOn('XY')
+  |> startProfileAt([-10, 10], %)
+  |> line([20, 0], %)
+  |> line([0, -20], %, $seg01)
+  |> line([-20, 0], %)
+  |> lineTo([profileStartX(%), profileStartY(%)], %)
+  |> close(%)
+const extrude001 = extrude(-15, sketch001)
+  |> fillet({ radius: 3, tags: [seg01] }, %)`
+
+    await runModifyAstWithFilletAndTagTest(
+      code,
+      segmentSnippets,
+      radiusValue,
+      expectedCode
+    )
+  })
+  it('should add a fillet to the 2 segments of a single extrusion', async () => {
+    const code = `const sketch001 = startSketchOn('XY')
+  |> startProfileAt([-10, 10], %)
+  |> line([20, 0], %)
+  |> line([0, -20], %)
+  |> line([-20, 0], %)
+  |> lineTo([profileStartX(%), profileStartY(%)], %)
+  |> close(%)
+const extrude001 = extrude(-15, sketch001)`
+    const segmentSnippets = ['line([20, 0], %)', 'line([-20, 0], %)']
+    const radiusValue = 3
+    const expectedCode = `const sketch001 = startSketchOn('XY')
+  |> startProfileAt([-10, 10], %)
+  |> line([20, 0], %, $seg01)
+  |> line([0, -20], %)
+  |> line([-20, 0], %, $seg02)
+  |> lineTo([profileStartX(%), profileStartY(%)], %)
+  |> close(%)
+const extrude001 = extrude(-15, sketch001)
+  |> fillet({ radius: 3, tags: [seg01] }, %)
+  |> fillet({ radius: 3, tags: [seg02] }, %)`
+
+    await runModifyAstWithFilletAndTagTest(
+      code,
+      segmentSnippets,
+      radiusValue,
+      expectedCode
+    )
+  })
```

Comment:
This might already be covered by an existing test so ignore me if it's already covered, but maybe one more tests that like
```ts
it('It should be able to add a new fillet when the extrude variable previously had an fillet', async () => {
    const code = `const sketch001 = startSketchOn('XY')
  |> startProfileAt([-10, 10], %)
  |> line([20, 0], %)
  |> line([0, -20], %, $seg01)
  |> line([-20, 0], %)
  |> lineTo([profileStartX(%), profileStartY(%)], %)
  |> close(%)
const extrude001 = extrude(-15, sketch001)
  |> fillet({ radius: 3, tags: [seg01] }, %)` // <--- one fillet already there on input code
    const segmentSnippets = ['line([20, 0], %)']
    const radiusValue = 3
    const expectedCode = `const sketch001 = startSketchOn('XY')
  |> startProfileAt([-10, 10], %)
  |> line([20, 0], %)
  |> line([0, -20], %, $seg01)
  |> line([-20, 0], %)
  |> lineTo([profileStartX(%), profileStartY(%)], %)
  |> close(%)
const extrude001 = extrude(-15, sketch001)
  |> fillet({ radius: 3, tags: [seg01] }, %)
  |> fillet({ radius: 3, tags: [seg02] }, %)` // <-- able to add a new one

    await runModifyAstWithFilletAndTagTest(
      code,
      segmentSnippets,
      radiusValue,
      expectedCode
    )
  })
```
☝️ not tested so probably wouldn't work out the box
I know this case of testing another fillet can be added when there already is one is already pretty much covered just by how you said this this works by just looping with `addFillet` but no harm in having an extra check. Up to you.